examples/zoom-provider/server/zoomprovider.js

```diff
@@ -0,0 +1,116 @@
+const request = require('request')
+const BASE_URL = 'https://zoom.us/v2'
+
+/**
+ * an example of a custom provider module. It implements @uppy/companion's Provider interface
+ */
+class ZoomProvider {
+  constructor (options) {
+    this.authProvider = ZoomProvider.authProvider
+  }
+
+  static get authProvider () {
+    return 'zoom'
+  }
+
+  // /users/{userId}/recordings
+  list ({ token, query }, done) {
+    const path = '/users/me/recordings'
+    const options = {
+      url: `${BASE_URL}${path}`,
+      method: 'GET',
+      json: true,
+      headers: {
+        Authorization: `Bearer ${token}`
+      }
+    }
+
+    request(options, (err, resp, body) => {
+      if (err) {
+        console.log(err)
+        done(err)
+        return
+      }
+      done(null, this._adaptData(body))
+    })
+  }
+
+  download ({ id, token }, onData) {
+    const downloadUrl = id
```

Comment:
I assumed this gets invoked when files get downloaded from source / uploaded to the destination. What's the best way to test this locally? I"m guessing i would need to attach a file destination or processing plugin and confirm this works independently, is this correct?

Reply:
>Likewise, I wasn't sure what the best way to pass additional data back to the UI and back to the download function heres was. The download URL gets returns as part of the list request, so I just assigned it to the ID parameter since it appeared that this was getting passed around automatically.

you can send an actual ID (if it exists in zoom), and then make a call within this `download(...)` function which will fetch the corresponding URL, and then proceed to actually download it.

Reply:
> I assumed this gets invoked when files get downloaded from source / uploaded to the destination. What's the best way to test this locally? I"m guessing i would need to attach a file destination or processing plugin and confirm this works independently, is this correct?

I think my comment [here](https://github.com/transloadit/uppy/pull/2342#issuecomment-650323365) might have already answered this question, but please let me know if this is not the case